crates/agglayer-node/src/rpc/mod.rs

```diff
@@ -136,22 +136,26 @@ where
 
         if !self.kernel.check_rollup_registered(tx.tx.rollup_id) {
             // Return an invalid params error if the rollup is not registered.
-            return Err(Error::rollup_not_registered(tx.tx.rollup_id));
+            return Err(Error::network_not_registered(tx.tx.rollup_id));
         }
 
+        self.kernel.verify_signature(&tx).await.map_err(|e| {
+            error!(error = %e, hash, "Failed to verify the signature of transaction {hash}: {e}");
+            Error::signature_mismatch(e)
+        })?;
+
+        agglayer_telemetry::VERIFY_SIGNATURE.add(1, metrics_attrs);
+
+        // Reserve a rate limiting slot.
+        let guard = self
+            .kernel
+            .rate_limiter()
+            .reserve_send_tx(tx.tx.rollup_id, tokio::time::Instant::now())?;
```

Comment:
Just to make sure, is the `tx.tx.rollup_id` here the same ID we want to be specified in the config file? I vaguely remember there being distinction between `network_id` and `rollup_id`.

Reply:
`network_id` and `rollup_id` are the same things unless `network_id = 0` which is L1
And yes this is the same ID as in the config, but I would like to not rely on the config for defining the "participant" on the agglayer ^^